examples/go-client/main.go

```diff
@@ -18,26 +18,56 @@ package main
 
 import (
 	"context"
-	"fmt"
 	"log"
+
+	"github.com/gravitational/teleport/api/client"
+	"github.com/gravitational/teleport/api/types"
+
+	"github.com/pborman/uuid"
 )
 
 func main() {
+	ctx := context.Background()
 	log.Printf("Starting Teleport client...")
-	client, err := connectClient()
+
+	creds, err := client.ProfileCreds()
 	if err != nil {
-		log.Fatalf("Failed to create client: %v", err)
+		log.Fatalf("Failed to create Credentials: %v", err)
 	}
-	defer client.Close()
 
-	ctx := context.Background()
+	clt, err := client.New(client.Config{
+		Addrs:       []string{"proxy.example.com:3025"},
+		Credentials: creds,
+	})
+	if err != nil {
+		log.Fatalf("Failed to create client: %v", err)
+	}
+	defer clt.Close()
 
-	fmt.Println("")
-	roleCRUD(ctx, client)
+	// Create a new access request for api-admin to use the admin role.
+	accessReq, err := types.NewAccessRequest(uuid.New(), "access-admin", "admin")
+	if err != nil {
+		log.Fatalf("Failed to make new access request: %v", err)
+	}
+	if err = clt.CreateAccessRequest(ctx, accessReq); err != nil {
+		log.Fatalf("Failed to create access request: %v", err)
+	}
+	log.Printf("Created access request: %v", accessReq)
 
-	fmt.Println("")
-	tokenCRUD(ctx, client)
+	defer func() {
+		if err = clt.DeleteAccessRequest(ctx, accessReq.GetName()); err != nil {
+			log.Fatalf("Failed to delete access request: %v", err)
+		}
+		log.Println("Deleted access request")
+	}()
 
-	fmt.Println("")
-	accessWorkflow(ctx, client)
+	// Approve the access request.
+	if err = clt.SetAccessRequestState(ctx, types.AccessRequestUpdate{
+		RequestID: accessReq.GetName(),
+		State:     types.RequestState_APPROVED,
+	}); err != nil {
+		// We use a panic here to ensure the deferred function still runs.
```

Comment:
This example could be moved to a separate function returning an error to call from `main`. This would not be needed.